[Builder] Remove ContainerUpdateCmdOnBuild

dnephin opened the pull request:
`ContainerUpdateCmdOnBuild` is an unnecessary extra method on the `builder.Backend` interface. It seems to have been added (39343b86182b4e997dc991645729ae130bd0f5f2) as a way to deal with really old code in the builder that messed with container config after creating the container.

Instead we can set the entrypoint to the `{""}` token, which maybe didn't exist when this code was first written.

Also cleans up setting of `b.runConfig.Image = b.image`, which was does at random places. I believe the correct place for this operation is between dispatch steps. With further refactoring there should be a bunch of other similar operations that happen at this time, as fields are removed from the `Builder` struct.

dnephin committed:
Remove ContainerUpdateCmdOnBuild, it does nothing.

Set a blank entrypoint to preserve the old behaviour.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

dnephin commented:
It looks like windowsRS1 is failing because of this, but I'm not sure why

tonistiigi commented:
@dnephin The win/linux difference might be caused by windows busybox image defining an entrypoint while linux one doesn't.

dnephin commented:
I think that's right. I added some debug and tried to override the entrypoint , but that cause a couple other tests to fail. I might rebase this on top of #32772 which I think would make the code a lot more obvious.

dnephin committed:
Fix setting b.runConfig.Image at arbitrary places.

Previously this value was set at some point attrbitrarily between when it was updated and when it was going to be used next.

Instead always set it as the last step of dispatch.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

dnephin committed:
Fix run with entrypoint in base image

Update a test to use a base image with entrypoint to that the linux build
has at least one test that behaves like all the windows tests.

Signed-off-by: Daniel Nephin <dnephin@docker.com>

tonistiigi reviewed (approved):
LGTM

vdemeester reviewed (approved):
LGTM 🦁
